Documentation updated: Hosting from a sub path section added (#14505)
keycloak/keycloak: docs/guides/src/main/server/reverseproxy.adoc

If this header is incorrectly configured, rogue clients can set this header and trick Keycloak into thinking the client is connected from a different IP address than the actual address.
This precaution can be more critical if you do any deny or allow listing of IP addresses.

> == Hosting keyclock on subpath

Reviewer: ```suggestion
== Different context-path on reverse proxy
```

> When Keycloak is hosted behind a reverse proxy, it is usually served from a subpath (not from rool url), for this to work properly Keyclock should be configured to serve resources from a subpath. Use `KC_HTTP_RELATIVE_PATH` for this, e.g. if you want to set Keyclock to host from subpath `/auth` then set this environment variable to `/auth`, see details https://www.keycloak.org/server/all-config?f=build

Reviewer: ```suggestion
Keycloak assumes it is exposed through the reverse proxy under the same context path as Keycloak is configured for. By default Keycloak is exposed through the root (`/`).

You can use `hostname-path` or `hostname-url` in these cases, or alternatively you can also change the context path of Keycloak itself to match the context path for the reverse proxy using the `http-relative-path` option.

When Keycloak is hosted behind a reverse proxy, it is usually served from a subpath (not from rool url), for this to work properly Keyclock should be configured to serve resources from a subpath. Use `KC_HTTP_RELATIVE_PATH` for this, e.g. if you want to set Keyclock to host from subpath `/auth` then set this environment variable to `/auth`, see details https://www.keycloak.org/server/all-config?f=build
```

If this header is incorrectly configured, rogue clients can set this header and trick Keycloak into thinking the client is connected from a different IP address than the actual address.
This precaution can be more critical if you do any deny or allow listing of IP addresses.

== Different context-path on reverse proxy

> Keycloak assumes it is exposed through the reverse proxy under the same context path as Keycloak is configured for. By default Keycloak is exposed through the root (`/`).
> You can use `hostname-path` or `hostname-url` in these cases, or alternatively you can also change the context path of Keycloak itself to match the context path for the reverse proxy using the `http-relative-path` option. e.g. to configure it using `http-relative-path`, if you want to set Keyclock to host from subpath `/auth` then set `http-relative-path` to `/auth`, see details https://www.keycloak.org/server/all-config?f=build

Reviewer: ```suggestion
Keycloak assumes it is exposed through the reverse proxy under the same context path as Keycloak is configured for. By default Keycloak is exposed through the root (`/`), which means it expects to be exposed through the reverse proxy on `/` as well.

You can use `hostname-path` or `hostname-url` in these cases, for example using `--hostname-path=/auth` if Keycloak is exposed through the reverse proxy on `/auth`.
 
Alternatively you can also change the context path of Keycloak itself to match the context path for the reverse proxy using the `http-relative-path` option, which will change the context-path of Keycloak itself to match the context path used by the reverse proxy.
```

If this header is incorrectly configured, rogue clients can set this header and trick Keycloak into thinking the client is connected from a different IP address than the actual address.
This precaution can be more critical if you do any deny or allow listing of IP addresses.

== Different context-path on reverse proxy

Keycloak assumes it is exposed through the reverse proxy under the same context path as Keycloak is configured for. By default Keycloak is exposed through the root (`/`), which means it expects to be exposed through the reverse proxy on `/` as well.
You can use `hostname-path` or `hostname-url` in these cases, for example using `--hostname-path=/auth` if Keycloak is exposed through the reverse proxy on `/auth`.
 

> Alternatively you can also change the context path of Keycloak itself to match the context path for the reverse proxy using the `http-relative-path` option, which will change the context-path of Keycloak itself to match the context path used by the reverse proxy. see details of configuration options at https://www.keycloak.org/server/all-config?f=build

Author: Done, wasn't aware of `includedOption` syntax, but that is nice feature, I hope I did it in right way.
